tasks/bundle.js

```diff
@@ -28,48 +29,76 @@ try {
 }
 catch(e) {
     throw new Error([
-        'build/ is missing a or more files',
+        'build/ is missing one or more files',
         'Please run `npm run preprocess` first'
     ].join('\n'));
 }
 
 
 // Browserify plotly.js
-browserify(constants.pathToPlotlyIndex, {
-    debug: DEV,
+_bundle(constants.pathToPlotlyIndex, constants.pathToPlotlyDist, {
     standalone: 'Plotly',
-    transform: [compressAttributes]
-})
-.bundle(function(err, buf) {
-    if(err) throw err;
-
-    // generate plotly.min.js
-    if(!DEV) {
-        fs.writeFile(
-            constants.pathToPlotlyDistMin,
-            UglifyJS.minify(buf.toString(), constants.uglifyOptions).code
-        );
-    }
-})
-.pipe(fs.createWriteStream(constants.pathToPlotlyDist));
-
+    debug: DEV,
+    pathToMinBundle: constants.pathToPlotlyDistMin
+});
 
 // Browserify the geo assets
-browserify(constants.pathToPlotlyGeoAssetsSrc, {
+_bundle(constants.pathToPlotlyGeoAssetsSrc, constants.pathToPlotlyGeoAssetsDist, {
     standalone: 'PlotlyGeoAssets'
-})
-.bundle(function(err) {
-    if(err) throw err;
-})
-.pipe(fs.createWriteStream(constants.pathToPlotlyGeoAssetsDist));
-
+});
 
 // Browserify the plotly.js with meta
-browserify(constants.pathToPlotlyIndex, {
-    debug: DEV,
-    standalone: 'Plotly'
-})
-.bundle(function(err) {
-    if(err) throw err;
-})
-.pipe(fs.createWriteStream(constants.pathToPlotlyDistWithMeta));
+_bundle(constants.pathToPlotlyIndex, constants.pathToPlotlyDistWithMeta, {
+    standalone: 'Plotly',
+    debug: DEV
+});
+
+// Browserify the plotly.js partial bundles
+constants.partialBundlePaths.forEach(function(pathObj) {
+    _bundle(pathObj.index, pathObj.dist, {
+        standalone: 'Plotly',
+        debug: DEV,
+        pathToMinBundle: pathObj.distMin
+    });
+});
+
+function _bundle(pathToIndex, pathToBundle, opts) {
```

Comment:
Just a thought: what about moving this bundle function to its own file and exporting it? Then `tasks/bundle.js` would require and *use* `_bundle`. Which is all the same, except that it would make room for a standalone bundler script to also just use this function. As long as there were a stable, guaranteed location for this script, a standalone bundler repo would probably just require this script and not have to reimplement it.

Other alternative is that a standalone bundler would implement this and get required by `plotly.js` as a dev dependency to do the bundling.